apps/web/src/app/challenge/[slug]/right-wrapper.tsx

```diff
@@ -34,8 +35,11 @@ export function RightWrapper({ track, challenge }: Props) {
     slug?: string,
   ) {
     const query = slug ? `&slug=${slug}` : '';
+
     isSuccessful &&
       router.push(`/challenge/${challenge.slug}/submissions/${submissionId}?success=true${query}`);
+    isSuccessful && session?.user?.id &&
```

Comment:
if the submission was successful, recalculate the badges as they may have earned one. Fairly dummy/naive implementation, as we may be able to tell ahead of time if its likely they completed a badge, but good for now?